insights/core/__init__.py

```diff
@@ -434,6 +434,19 @@ def _parse_line(self, line):
         """
         return {'raw_message': line}
 
+    def _valid_search(self, s):
+        """
+        Check this given `s`, it must be a string or a list of strings.
+        Otherwise, a TypeError will be raised.
+        """
+        if isinstance(s, str):
+            return lambda l: s in l  # noqa: E731
+        elif (isinstance(s, list) and len(s) > 0 and
+                    all(isinstance(w, str) for w in s)):
+            return lambda l: all(w in l for w in s)  # noqa: E731
```

Comment:
Is there a reason to have the `noqa` on these two lines? I didn't see any flake8 errors when they were removed.